core/partitioning/partitioning.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <vector>
+
+#include "core/conversion/conversion.h"
+#include "torch/csrc/jit/ir/ir.h"
+
+namespace trtorch {
+namespace core {
+namespace partitioning {
+
+torch::jit::Value* getOrAddInputForValue(
+    torch::jit::Value* old_value,
+    std::shared_ptr<torch::jit::Graph>& graph,
+    std::unordered_map<torch::jit::Value*, torch::jit::Value*>& old_to_new);
+
+torch::jit::Node* cloneNode(
+    torch::jit::Node* node,
+    std::shared_ptr<torch::jit::Graph>& graph,
+    std::unordered_map<torch::jit::Value*, torch::jit::Value*>& old_to_new);
+
+struct SegmentedBlock {
+ public:
+  enum SegmentedBlockTarget {
+    kTorch,
+    kTensorRT,
+  };
+
+  SegmentedBlock() = default;
+
+  SegmentedBlock(SegmentedBlockTarget blk_target) : target_(blk_target), g_(std::make_shared<torch::jit::Graph>()) {}
+
+  SegmentedBlock(SegmentedBlockTarget blk_target, const std::vector<torch::jit::Node*>& nodes)
+      : target_(blk_target), g_(std::make_shared<torch::jit::Graph>()) {
+    for (auto& node : nodes) {
+      appendNode(node);
+    }
+  }
+
+  SegmentedBlock(SegmentedBlockTarget blk_target, std::shared_ptr<torch::jit::Graph> g) : target_(blk_target), g_(g) {}
+
+  enum SegmentedBlockTarget target() {
+    return target_;
+  }
+
+  void appendNode(torch::jit::Node* n) {
+    cloneNode(n, g_, old_to_new_);
+  }
+
+  void registerInputs() {
+    for (auto& value : g_->inputs()) {
+      inputs_.push_back(old_to_new_[value]);
+    }
+  }
+
+  void registerOutput(torch::jit::Value* raw_input) {
+    outputs_.push_back(raw_input);
+    g_->registerOutput(old_to_new_[raw_input]);
+  }
+
+  torch::jit::Block* block() {
+    return g_->block();
+  }
+
+  c10::ArrayRef<torch::jit::Value*> inputs() {
+    return g_->inputs();
+  }
+
+  c10::ArrayRef<torch::jit::Value*> outputs() {
+    return g_->outputs();
+  }
+
+  const std::vector<torch::jit::Value*>& raw_inputs() const {
+    return inputs_;
+  }
+
+  const std::vector<torch::jit::Value*>& raw_outputs() const {

```

Comment:
What would be the difference between raw_inputs() and input() values in practice ? Do we need both variants ?

Reply:
raw_inputs() would be the torch::jit::Value that in the lowering global graph that is used as an input in our mini graph, when we construct the mini-graphs these Values in mini-graph will change because we are constructing new graph. We need them both because when we stitch mini-graph together, we need the raw_inputs to find the mappings, and we use inputs() to run the segments inferences and do something like replaceAllUsesWith() for mini-graph inputs.